tests/mypy_test.py

```diff
@@ -44,6 +46,24 @@
     print_error("Cannot import mypy. Did you install it?")
     sys.exit(1)
 
+# We need to work around a limitation of tempfile.NamedTemporaryFile on Windows
+# For details, see https://github.com/python/typeshed/pull/13620#discussion_r1990185997
+# Python 3.12 added a workaround with `tempfile.NamedTemporaryFile("w+", delete_on_close=False)`
+if sys.platform != "win32":
```

Comment:
If any contributor runs this on win, it will be a `NameError` to access `_named_temporary_file`. I guess we have to explicitly say that `win32` platform is not supported for running tests.

Reply:
On Windows, `_named_temporary_file` is defined in the `else` branch. The idea is to work around a limitation in Windows.

Maybe this would be more clear if the condition was inverted:

```python
if sys.platform == "win32":
    ...
else:
    ...
```

Reply:
I totally missed that, thank you!

Reply:
I originally had a "if python 3.12" branch first. But figured it's just introducing a different behaviour for another set of variables, so I removed it in favor of a simple comment.

It made more sense then to have the two different   functools.partial first, then the bigger block.

But now I don't mind flipping the condition for readability.